### DISCUSSION TITLE
Telegram integration and ftp upload.

### DISCUSSION THREAD

**Discussion by aliSYB** [beeyev/Mikrotik-RouterOS-automatic-backup-and-update#8] [Ideas]
Hi,
thank you for the great job done with the script.
are you interested in Telegram integration?
i have modified your code to send notification to Telegram, and i'm working on ftp upload of the backup files.

**Comment by beeyev** [comment]
Hello, and thank you for contributing, of course I am interested. 
Please contact me using my [email](https://github.com/beeyev/Mikrotik-RouterOS-automatic-backup-and-update/blob/65e4ac4f5827738c988b04b853787d212e87c9a1/BackupAndUpdate.rsc#L11) (its written is the source)
Or you can attach your code here.

**Comment by TerAnYu** [comment]
A version of the backup may be interesting: https://github.com/TerAnYu/mikrotik_ftp_backup

**Reply by aliSYB** [reply]
this version is much simpler for ftp upload, no need for certificates.

## ftp upload of backup files
## modify below as needed.
## ftpUpload if true will upload, if false will not upload.
# ftp settings
:local ftpUpload		true;
:local ftpAddress		"ip address";
:local ftpPort			"ftp port";
:local ftpUser			"username";
:local ftpPassword		"password";
:local ftpDestination	"/ftpBackup";`

:if ($ftpUpload = true) do={
/tool fetch mode=ftp address="$ftpAddress" port="$ftpPort" user="$ftpUser" password="$ftpPassword"\
 src-path="$systemBackupDir/$backupFilename" dst-path="$ftpDestination/$backupFilename" upload=yes;
}

i'm also trying to do the same for email, as i don't want to configure the mikrotik email.
## Notification e-mail
## (Make sure you have configurated Email settings in Tools -> Email)
## SendEmailTo # Destination email address. Single address allowed.
## SendEmailCC # Send a copy to listed recipients. Multiple addresses allowed, use "," to separate entries
## UseTLS Possible values: yes, no, tls-only / If UseTLS="tls-only", port 465 will be used.
:local SendEmail 	 	true;
:local SendEmailFrom 	"sender@gmail.com";
:local PasswordMail 	"Password";
:local SmtpServer 		[:resolve "smtp.gmail.com"];
:local SendEmailTo 		"receiver@gmail.com";
:local SendEmailCC		"receiver1@gmail.com, receiver2@gmail.com";
:local SmtpPort 		587;
:local UseTLS 		 	"no";

:if ($SendEmail = true) do={
/tool e-mail send to=$SendEmailTo cc=$SendEmailCC server=$SmtpServer port=$SmtpPort\
 start-tls=$UseTLS user=$SendEmailFrom password=$PasswordMail from=$SendEmailFrom\
 subject=$mailSubject body=$mailBody file=$mailAttachments;
}

**Reply by TerAnYu** [reply]
certificates are required for sstp/openvpn/ikev2, without them the backup is not complete (recently it was necessary to restore several devices from the backup due to device failure and the inability to save information from flash)
